pkg/sidecar/util_posix.go

```diff
@@ -17,9 +17,56 @@ import (
 // When a new SVID is received on the updateChan, the SVID certificates
 // are stored in disk and a restart signal is sent to the proxy's process
 func (s *Sidecar) RunDaemon(ctx context.Context) error {
-	err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
-	if err != nil && !errors.Is(err, context.Canceled) {
-		return err
+	var wg sync.WaitGroup
+	ctx, cancel := context.WithCancel(ctx)
+	defer func() {
+		cancel()
+		wg.Wait()
+	}()
+
+	tasks := 0
+	if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
+		tasks++
+	}
+	if s.config.JSONFilename != "" {
+		tasks++
+	}
+	if s.config.JSONFilename != "" && s.config.JwtAudience != "" {
+		tasks++
+	}
+	wg.Add(tasks)
+
+	errch := make(chan error, tasks)
+	if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
+		go func() {
+			defer wg.Done()
+			errch <- workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+		}()
+	}
+	if s.config.JSONFilename != "" {
+		go func() {
+			defer wg.Done()
+			errch <- workloadapi.WatchJWTBundles(ctx, &JWTBundlesWatcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+		}()
+	}
+	if s.config.JSONFilename != "" && s.config.JwtAudience != "" {
+		go func() {
+			defer wg.Done()
+			s.updateJWTSVID(ctx, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+			errch <- nil
+		}()
+	}
+
```

Comment:
What do you think of structuring this section more like in here: https://github.com/spiffe/go-spiffe/blob/main/v2/examples/spiffe-watcher/main.go#L39-L59

Reply:
I think it's a good idea, it's more readable.